modules/tinymce/src/core/main/ts/api/html/DomParser.ts

```diff
@@ -97,21 +98,32 @@ type WalkerCallback = (node: AstNode) => void;
 const nonRcDataSpecialElements = new Set('style,xmp,iframe,noembed,noframes,script,noscript,plaintext'.split(','));
 const specialElements = new Set([ 'title', 'textarea', ...nonRcDataSpecialElements ]);
 
-const getPurifyConfig = (settings: DomParserSettings): Config => {
-  const config: Config = {
-    RETURN_DOM: true,
-    ALLOW_DATA_ATTR: true,
-    ALLOWED_URI_REGEXP: /^(?:(?:(?:f|ht)tps?|about|blob|file|mailto|tel|callto|cid|xmpp):|[^a-z]|[a-z+.\-]+(?:[^a-z+.\-:]|$))/i,
-  };
+const basePurifyConfig: Config & { RETURN_DOM: true } = {
+  RETURN_DOM: true,
+  ALLOW_UNKNOWN_PROTOCOLS: true,
+  // Deliberately ban all tags and attributes by default, and then un-ban them on demand in hooks
+  // #comment and #cdata-section are always allowed as they aren't controlled via the schema
+  ALLOWED_TAGS: [ '#comment', '#cdata-section' ],
+  ALLOWED_ATTR: []
+};
 
-  if (settings.allow_html_data_urls) {
-    config.ADD_DATA_URI_TAGS = [ 'a', 'img' ];
```

Comment:
I'd mentioned this on Neil's previous PR but this was definitely wrong, so I've fixed this and other issues with the allow url settings.